### PR TITLE
[JENKINS-43199, JENKINS-45057] Reliably close build log file

### PR DESCRIPTION
**Pull request by jglick**
Simpler alternative to #2953.

* JENKINS-43199, JENKINS-45057 - Bug - Reliably close build log file when using chained Build Listeners

@reviewbybees

### CODE REVIEW

File: core/src/main/java/hudson/model/Run.java [hunk 1]
```diff
@@ -1787,6 +1789,13 @@ protected final void execute(@Nonnull RunExecution job) {
                     listener.finished(result);
                     listener.closeQuietly();
                 }
+                if (logger != null) {
```
Comment (stephenc) [inline core/src/main/java/hudson/model/Run.java]: if `listener.finished(result)` throws an exception, the logger will not be closed

**Review by alexanderrtaylor — APPROVED** [review]
This LGTM 🐝 

I think @stephenc's comments are valid but that is not a net new issue(it would not be closed properly before the new changes as well)

File: core/src/main/java/hudson/model/Run.java (hunk 1 above)
Comment (jglick) [inline core/src/main/java/hudson/model/Run.java]: True, but that was already the case; this patch is merely improving the behavior when `listener.close()` does not call `logger.close()`.

File: core/src/main/java/hudson/model/Run.java [hunk 2]
```diff
@@ -1798,23 +1800,32 @@ protected final void execute(@Nonnull RunExecution job) {
             try {
                 getParent().logRotate();
             } catch (Exception e) {
-		LOGGER.log(Level.SEVERE, "Failed to rotate log",e);
-	    }
+                LOGGER.log(Level.SEVERE, "Failed to rotate log",e);
+            }
```
Comment (jglick) [inline core/src/main/java/hudson/model/Run.java]: #1790 introduced tabs.

**Review by stephenc — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]
@reviewbybees done

**Comment by oleg-nenashev** [comment]
I am integrating it in order to get the patch in 2.60.3 RC

**Pull request merged**